templates/react/packages/client/src/App.tsx

```diff
@@ -1,105 +1,46 @@
-import { useMUD } from "./MUDContext";
-
-const styleUnset = { all: "unset" } as const;
-
-export const App = () => {
-  const {
-    network: { tables, useStore },
-    systemCalls: { addTask, toggleTask, deleteTask },
-  } = useMUD();
-
-  const tasks = useStore((state) => {
-    const records = Object.values(state.getRecords(tables.Tasks));
-    records.sort((a, b) => Number(a.value.createdAt - b.value.createdAt));
-    return records;
-  });
+import { stash } from "./mud/stash";
+import { useRecords } from "@latticexyz/stash/react";
+import { AccountButton } from "@latticexyz/entrykit/internal";
+import { Direction } from "./common";
+import mudConfig from "contracts/mud.config";
+import { useMemo } from "react";
+import { GameMap } from "./game/GameMap";
+import { useWorldContract } from "./mud/useWorldContract";
+import { Synced } from "./mud/Synced";
+import { useSync } from "@latticexyz/store-sync/react";
+
+export function App() {
+  const players = useRecords({ stash, table: mudConfig.tables.app__Position });
+
+  const sync = useSync();
+  const worldContract = useWorldContract();
+  const onMove = useMemo(
+    () =>
+      sync.data && worldContract
+        ? async (direction: Direction) => {
+            const tx = await worldContract.write.app__move([mudConfig.enums.Direction.indexOf(direction)]);
+            await sync.data.waitForTransaction(tx);
+          }
+        : undefined,
+    [sync.data, worldContract],
+  );
```

Comment:
This still feels a little clunky to me. I tried to streamline this by doing the sync data check inside `useWorldContract` so that we could have one hook to return both `worldContract` and `waitForTransaction` (since you're often doing these together), but I was constantly battling TS depth issues where editing anything in `useWorldContract` caused it to spin due to the huge world ABI.

A potential next step to improve this would be better utils to call individual systems via their individual ABIs rather than a monolithic world contract, but wagmi doesn't make it easy to call using a separate client (session client, in our case).